### ISSUE TITLE
Community: Can't set permission values as decimals

### ISSUE THREAD

**Issue by mprakhov** [status-im/status-desktop#10413]
# Bug Report

## Description

Can't set permission values as decimals. This depends on the locale.

Note:
If I replace 
```
function numberFromLocaleString(num, locale = null) {
        locale = locale || Qt.locale()
...
}
```
to 
```
function numberFromLocaleString(num, locale = null) {
        locale = Qt.locale()
...
}
```
everything works fine

## Steps to reproduce

Change locale format

https://user-images.githubusercontent.com/117639195/234028673-ef80d1a5-c833-46e2-9f48-64fc25576c21.mov

#### Expected behavior

The user should be able to set decimal values independent of his locale

#### Actual behavior

The user can't set decimal value due to locale

### Additional Information

- Status desktop version: 0.11.0
- Operating System: macos

**Comment by jrainville** [comment]
I think this might be fixed in master cc @alexandraB99

**Comment by mprakhov** [comment]
@jrainville nope, it is reproducible on master too.

**Comment by caybro** [comment]
CC @micieslak I think you worked on this? It looks like we're passing a wrong locale to the method

**Comment by caybro** [comment]
OK, so this is on "purpose", and was done as part of this PR: https://github.com/status-im/status-desktop/pull/10169

TLDR, we only allow "en_US" locale on input, that's why you have to use a dot `.` as decimal separator

**Comment by caybro** [comment]
The relevant code pieces:
```qml
AmountInput {
        id: amountInput

        locale: LocaleUtils.userInputLocale
}
```

and LocaleUtils:
```qml
readonly property var userInputLocale: Qt.locale("en_US")
```

**Comment by caybro** [comment]
Can we close this @John-44 or do you see it as a relevant issue? (I warned against hardcoding a decimal separator everywhere :) It's totally counter-intuitive... most countries do _not_ use `.` (period, full stop)

**Comment by mprakhov** [comment]
@caybro I can't use `.` as it is not passing validation. Clicking `.` does nothing for me due to
```
if (!allowDecimals)
            text = text.replace(root.locale.decimalPoint, "")

        if(text.length === 0) {
            d.amount = 0
            root.validationError = ""
            return
        }
```
So the only option to set a decimal value - is to change locale settings on my machine 😢

**Comment by caybro** [comment]
> @caybro I can't use `.` as it is not passing validation. Clicking `.` does nothing for me due to
> 
> ```
> if (!allowDecimals)
>             text = text.replace(root.locale.decimalPoint, "")
> 
>         if(text.length === 0) {
>             d.amount = 0
>             root.validationError = ""
>             return
>         }
> ```
> 
> So the only option to set a decimal value - is to change locale settings on my machine cry

Ah that's strange... I can use the `.` there to enter the decimal value

**Comment by mprakhov** [comment]
setting

![image](https://user-images.githubusercontent.com/117639195/234719333-4e11465e-cc25-4f37-a3e0-e32e20797bc1.png)

**Comment by caybro** [comment]
Works under linux...

![image](https://user-images.githubusercontent.com/5377645/234719444-645da304-5c5c-44ab-8bb3-af53ba14c2ac.png)

But still... we should fix this, and I mean properly :)

**Comment by caybro** [comment]
@mprakhov because the input locale is hardcoded to `en_US`, it now thinks that your decimal value "0,001" is in fact `1` because it treats the comma (`,`) as a _thousands_ separator, not _decimal_ separator as one would expect

Even more confusion ensues... :facepalm:

**Comment by caybro** [comment]
> I thought that when we last discussed this the conclusion was they we would just use "." as the decimal separator (for now at least) to simplify things (plus the anecdotal experience from team members is that folks who live in countries that don't use . as the decimal separator are used to using . as a decimal separator in apps and online anyhow e.g. it's become an informal international standard.
> 
> The only view I hold here is that for MVP we should do the simplest thing that works, as we have many other unrelated items that also need attention to get the MVP out of the door.
> 
> We could discuss at the Friday desktop team call tomorrow if that would be useful?

Yeah, let's discuss on Friday... we need to fix this issue, it's quite severe imho (not being able to input a _correct_ value, or misleading users: imagine someone wants to send `1.123 ETH` (one something) and sends `1123 ETH` instead :fearful:)

**Comment by caybro** [comment]
The actual fix is simple:
```qml
LocaleUtils.qml:
readonly property var userInputLocale: Qt.locale()
```

**Comment by caybro** [comment]
@dlipicar any input on this, what are we missing that we had to force this "workaround"?

**Comment by dlipicar** [comment]
Oh well, this was kind of a scream test and it seems people are already screaming. It seems the expected behavior for user input is highly user-dependent. We should perhaps add a setting and let the user decide if they want Status to have numbers (both input and what we display, for consistency) use the system locale, dot as decimal separator or comma as decimal separator. And apply that change on the next start so stuff doesn't explode :D 

As for this particular issue, I *think* the problem is that `AmountInput` uses by default `DoubleValidator` instead of `StatusFloatValidator`. Let me do a quick test.

**Comment by caybro** [comment]
> Oh well, this was kind of a scream test and it seems people are already screaming. It looks like the expected behavior for user input is highly user-dependent. We should perhaps add a setting and let the user decide if they want Status to have numbers (both input and what we display, for consistency) use either the system locale, dot as decimal separator or comma as decimal separator. And apply that change on the next start so stuff doesn't explode :D

Personally it would be fine with me to have this user-configurable but for the love of Pete, with the _correct_ (OS) defaults :)

> As for this particular issue, I _think_ the problem is that `AmountInput` uses by default `DoubleValidator` instead of `StatusFloatValidator`. Let me do a quick test.

Great, thanks!

**Comment by dlipicar** [comment]
ugh `AmountInput` is an `Input` and  `AmountInputWithCursor` is a `StatusInput`, which are completely different beasts.

Anyway, `DoubleValidator` has locale `userInputLocale` (`en_US`) set, but for some reason it's considering `.` as decimal separator invalid, so it seems to still be using the system locale? And apparently this only happens in macOS? I'll need some extra time to properly investigate this...

**Comment by caybro** [comment]
StatusFloatValidator looks broken to me, we are not passing the `locale` to the internal `DoubleValidator`

**Comment by dlipicar** [comment]
Not sure why, but we're not really using the `DoubleValidator` there as far as I understand. We get an instance only to get default top and bottom values ¯\_(ツ)_/¯
The actual check gets done in the `validate` function

**Comment by caybro** [comment]
> Not sure why, but we're not really using the `DoubleValidator` there as far as I understand. We get an instance only to get default top and bottom values ¯_(ツ)_/¯ The actual check gets done in the `validate` function

Yeah but then the `top` and `bottom` are inherently wrong (or can be)

**Comment by dlipicar** [comment]
> > Not sure why, but we're not really using the `DoubleValidator` there as far as I understand. We get an instance only to get default top and bottom values ¯_(ツ)_/¯ The actual check gets done in the `validate` function
> 
> Yeah but then the `top` and `bottom` are inherently wrong (or can be)

hmm that shouldn't be locale-dependent, they're `real` values

**Comment by caybro** [comment]
> Anyway, `DoubleValidator` has locale `userInputLocale` (`en_US`) correctly set, but for some reason it's considering `.` as decimal separator invalid, so it seems to still be using the system locale? And apparently this only happens in macOS? I'll need some extra time to properly investigate this...

Hmm, might be a macOS specific issue; does it recognize the `en_US` locale? ;)

**Comment by caybro** [comment]
@dlipicar ping? can we just set `readonly property var userInputLocale: Qt.locale()` and get rid of this?

**Comment by iurimatias** [comment]
not ready for 0.12, moving to 0.13

**Comment by iurimatias** [comment]
@dlipicar could you do a PR for the RC for this please
